plugin/persistence/persistence.view.js

```diff
@@ -0,0 +1,191 @@
+			canExecute: () => !!this.title && this.isUniqueTitle(this.title)
+		});
+
+		this.edit = {
+			enter: new Command({
+				source: 'persistence.view',
+				execute: item => {
+					item = item || this.items.find(this.isActive);
+					if (!item) {
+						return false;
+					}
+					this.state.editItem = item;
+					this.state.oldValue = clone(item);
+					return true;
+				},
+				canExecute: () => this.state.editItem === null
+			}),
+			commit: new Command({
+				source: 'persistence.view',
+				shortcut: 'enter',
+				execute: item => {
+					item = item || this.state.editItem;
+					if (!this.title || !this.isUniqueTitle(item.title)) {
+						this.edit.cancel.execute();
+						return false;
+					}
+					item.modified = Date.now();
+					model.persistence()
+						.storage
+						.setItem(this.storageKey, this.items);
+					this.state.editItem = null;
+					return true;
+				},
+				canExecute: () => this.state.editItem !== null
+			}),
+			cancel: new Command({
+				source: 'persistence.view',
+				shortcut: 'escape',
+				execute: () => {
+					if (this.state.editItem !== null) {
+						const index = this.items.indexOf(this.state.editItem);
+						this.items.splice(index, 1, this.state.oldValue);
+						this.state.oldValue = null;
+						this.state.editItem = null;
+					} else {
+						this.closeEvent.emit();
+					}
+					this.cancelEvent.emit();
+					return true;
+				}
+			})
+		};
+
+		this.load = new Command({
+			source: 'persistence.view',
+			execute: item => this.persistenceService.load(item.model)
+		});
+
+		this.remove = new Command({
+			source: 'persistence.view',
+			execute: item => {
+				const index = this.items.indexOf(item);
+				if (index >= 0) {
+					this.items.splice(index, 1);
+
+					this.model.persistence()
+						.storage
+						.setItem(this.storageKey, this.items);
+					return true;
+				}
+				return false;
+			}
+		});
+
+		this.setDefault = new Command({
+			source: 'persistence.view',
+			execute: item => {
+				const index = this.items.indexOf(item);
+				if (index === -1) {
+					return false;
+				}
+
+				if (item.isDefault) {
+					item.isDefault = false;
+				} else {
+					this.items.forEach(i => i.isDefault = false);
+					item.isDefault = true;
+				}
+				this.items.splice(index, 1, item);
+
+				this.model.persistence()
+					.storage
+					.setItem(this.storageKey, this.items);
+				return true;
+			}
+		});
+
+		const commandManager = new CommandManager();
+		const shortcut = new Shortcut(new ShortcutDispatcher());
+
+		this.keyDown = e => shortcut.keyDown(e);
+
+		shortcut.register(commandManager, [
+			this.edit.enter,
+			this.edit.commit,
+			this.edit.cancel
+		]);
+	}
+
+	isActive(item) {
+		return JSON.stringify(item.model) === JSON.stringify(this.persistenceService.save()); // eslint-disable-line angular/json-functions
+	}
+
+	stringify(item) {
+		const model = item.model;
+		const targets = [];
+		const settings = this.model.persistence().settings;
+
+		for (let key in settings) {
```

Comment:
do we want to check all settings properties there or own?

Reply:
all iterable properties exactly